app/services/schedule_data_manager_service.rb

```diff
@@ -103,10 +105,26 @@ def find_temperatures
     # Only use temperature data within lower datetime bounds of school meter readings
     return cached_temperatures unless use_date_bounded_schedule_data?
 
-    dates_to_remove = cached_temperatures.keys.select { |date| date < @school.minimum_reading_date }
+    school_minimum_reading_date = school_minimum_reading_date_with_temperature_days_offset
+    dates_to_remove = cached_temperatures.keys.select { |date| date < school_minimum_reading_date }
     cached_temperatures.remove_dates!(*dates_to_remove)
   end
 
+  def school_minimum_reading_date_with_temperature_days_offset
+    @school.minimum_reading_date - temperature_days_offset.days
```

Comment:
Note: we still need to change this to fix the other error I noted. That could be done in a separate PR if you prefer.